app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala

```diff
@@ -276,9 +277,34 @@ class EmailController @Inject() (val helpToSaveService:          HelpToSaveServi
       def ifDE = {
         _: HTSSession ⇒
           {
-            doUpdate(HTSSession(None, None, None, false))(
-              SeeOther(routes.EmailController.verifyEmail().url)
-            )
+            doUpdate(HTSSession(None, None, None, false)) {
+              htsContext.userDetails.fold[Future[Result]](
```

Comment:
this is very similar to the behaviour under `confirmEmail` - would it be possible to return `SeeOther(routes.EmailController.confirmEmail()` here to re-use some of the logic there?